### PR TITLE
Add changes to dynamically update discount percentage

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/38154

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Verify that `Renew annually save %` show the right amount for different regions i.e. US, CA, FR, ES, AU, etc.
**Calculation for % should be as below,** 
Yearly price = monthly price * 12
Actual yearly price = yearly price of the subscription
Discount = Yearly price - Actual yearly price
Discount Percentage = (Discount / Yearly price) × 100

We show the round value of `Discount Percentage`. i.e. 16.3% => 16% and 16.6% => 17

### CODE REVIEW

**Review by tapanmodh — APPROVED** [review]

File: android/java/org/chromium/chrome/browser/billing/InAppPurchaseWrapper.java [hunk 1]
```diff
@@ -532,6 +532,25 @@ public String getFormattedFullProductPrice(ProductDetails productDetails) {
...
+        if (monthlyPricingPhase != null && yearlyPricingPhase != null) {
+            double yearlyPrice =
+                    ((double) monthlyPricingPhase.getPriceAmountMicros() / MICRO_UNITS) * 12;
+            double yearlyActualPrice =
+                    ((double) yearlyPricingPhase.getPriceAmountMicros() / MICRO_UNITS);
+            if (yearlyPrice == 0) {
+                return 0;
+            }
+            double percentage = (100 - (yearlyActualPrice * 100) / yearlyPrice);
+            Log.e("barve_vpn", (int) Math.round(percentage) + "");
```
[the remark this answers is not in the document]
Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/billing/InAppPurchaseWrapper.java]: should be resolved with new commit.

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by kjozwiak** [comment]
Asked @srirambv @GeetaSarvadnya and @btlechowski to run through https://github.com/brave/brave-core/pull/23500#issue-2285068639 which will cover `India` & `Poland`. @hffvld also went through the above via `US`.

### `India`

#### `Case #1`

```
Yearly price (11880) - Actual yearly price (9800) = Discount (2080)
(Discount [2080] - Yearly price[11880])/100 => 0.17508/100=> 17.5 => 18% (Discount %)
```

image

#### `Case #2`

Screenshot_20240509_124019 (1)

```
Yearly price = monthly price * 12
Yearly price = 990 * 12 = 11880

Actual yearly price = yearly price of the subscription
Actual yearly price = 9800

Discount = Yearly price - Actual yearly price
Discount = 11880 - 9800 = 2080

Discount Percentage = (Discount / Yearly price) × 100
Discount Percentage = (2080/11880) *100 = 17.5
```

### `Poland`

```
The percentage discount is: 16.65% so 17%
```

image (1)

### `USA`

```
Monthly price * 12 = Yearly price
$9.99 * 12 = $119.88

Actual yearly price = yearly price of the subscription
$99.99

Yearly price - Actual yearly price = Discount
$119.88 - $99.99 = $19.89

(Discount / Yearly price) × 100 = Discount Percentage
($19.89 / $119.88) * 100 = 16.59%
```

Screen_09-31-55

**Comment by srirambv** [comment]
Verified new pricing is shown for India on nightly
image

**Pull request merged**